### PR TITLE
Case 21226: Avatar Exporter v0.3.1 - export Unity material data to fst

### PR DESCRIPTION
**Pull request by dback2**
https://highfidelity.manuscript.com/f/cases/21226/

New Avatar Exporter package:
https://github.com/highfidelity/hifi/raw/0e1dbded229c5202806de742506c9b772d2d0046/tools/unity-avatar-exporter/avatarExporter.unitypackage

### CODE REVIEW

File: tools/unity-avatar-exporter/Assets/Editor/AvatarExporter.cs [hunk 1]
```diff
@@ -255,15 +266,67 @@ class BoneTreeNode {
...
+            }
+            json += "\"roughness\": " + roughness + ", ";
+            if (!string.IsNullOrEmpty(roughnessMap)) {
+                json += "\"roughnessMap\": \"" + roughnessMap + "\", ";
+            }
+            if (!string.IsNullOrEmpty(normalMap)) {
+                json += "\"normalMap\": \"" + normalMap + "\", ";
+            }
+            if (!string.IsNullOrEmpty(heightMap)) {
+                json += "\"heightMap\": \"" + heightMap + "\", ";
```
Comment (SamGondelman) [inline tools/unity-avatar-exporter/Assets/Editor/AvatarExporter.cs]: we don't have height maps

File: tools/unity-avatar-exporter/Assets/Editor/AvatarExporter.cs [hunk 2]
```diff
@@ -575,18 +652,46 @@ class BoneTreeNode {
...
+            string materialJson = "{ ";
+            foreach (var materialData in materialDatas) {
+                // if this is the only material in the mapping and it is the default name No Name mapped to No Name, 
+                // then the avatar has no embedded materials and this material should be applied to all meshes
+                string materialName = materialData.Key;
+                if (materialMappings.Count == 1 && materialName == DEFAULT_MATERIAL_NAME && 
+                    materialMappings[materialName] == DEFAULT_MATERIAL_NAME) {
+                    materialJson += "\"all\": ";
+                } else {
+                    materialJson += "\"[mat::" + materialName + "]\": ";
```
Comment (SamGondelman) [inline tools/unity-avatar-exporter/Assets/Editor/AvatarExporter.cs]: you don't need the square brackets here

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17916/HighFidelity-Beta-PR15115-771eaa1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17916/HighFidelity-Beta-Interface-PR15115-771eaa1.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17916/HighFidelity-Beta-PR15115-771eaa1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17916/HighFidelity-Quest-Beta-PR15115-771eaa1.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17916/HighFidelity-Beta-PR15115-771eaa1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17916/HighFidelity-Beta-Interface-PR15115-771eaa1.exe

File: tools/unity-avatar-exporter/Assets/Editor/AvatarExporter.cs (hunk 2 above)
Comment (dback2) [reply tools/unity-avatar-exporter/Assets/Editor/AvatarExporter.cs]: Removed

File: tools/unity-avatar-exporter/Assets/Editor/AvatarExporter.cs (hunk 1 above)
Comment (dback2) [reply tools/unity-avatar-exporter/Assets/Editor/AvatarExporter.cs]: Ah ok sounded like we did from talking to other Sam about not having both a normal map and a height map, but removed it and the warning if it does have both.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17925/HighFidelity-Beta-PR15115-9cc1a7a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17925/HighFidelity-Beta-Interface-PR15115-9cc1a7a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17925/HighFidelity-Beta-PR15115-9cc1a7a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17925/HighFidelity-Beta-Interface-PR15115-9cc1a7a.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17925/HighFidelity-Beta-PR15115-9cc1a7a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17925/HighFidelity-Quest-Beta-PR15115-9cc1a7a.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17970/HighFidelity-Beta-PR15115-0e1dbde.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17970/HighFidelity-Quest-Beta-PR15115-0e1dbde.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17970/HighFidelity-Beta-PR15115-0e1dbde.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17970/HighFidelity-Beta-Interface-PR15115-0e1dbde.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17970/HighFidelity-Beta-PR15115-0e1dbde.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17970/HighFidelity-Beta-Interface-PR15115-0e1dbde.exe

**Pull request merged**